pkg/ingester/ingester_v2.go

```diff
@@ -1438,8 +1479,27 @@ func (i *Ingester) numSeriesInTSDB() float64 {
 	return float64(count)
 }
 
+// getOldestUnshippedBlockMetric returns the unix timestamp of the oldest unshipped block or
+// 0 if all blocks have been shipped.
+func (i *Ingester) getOldestUnshippedBlockMetric() float64 {
+	i.userStatesMtx.RLock()
+	defer i.userStatesMtx.RUnlock()
+
+	oldest := uint64(0)
+	for _, db := range i.TSDBState.dbs {
+		if ts := db.getOldestUnshippedBlockTime(); oldest == 0 || ts < oldest {
+			oldest = ts
+		}
+	}
+
+	return float64(oldest / 1000)
+}
+
 func (i *Ingester) shipBlocksLoop(ctx context.Context) error {
-	shipTicker := time.NewTicker(i.cfg.BlocksStorageConfig.TSDB.ShipInterval)
+	// We add a slight jitter to make sure that if the head compaction interval and ship interval are set to the same
+	// value they don't clash (if they both continuously run at the same exact time, the head compaction may not run
+	// because can't successfully cas the state).
```

Comment:
What do you mean by "cas the state"? Not sure if I'm confused about a term  I don't know or a typo ?

Reply:
By cas I mean "Compare And Swap" (we have a function called `casState()`). Maybe "CAS" (upper case) is more clear? Should I explode it to "compare and swap"?